### PR TITLE
[collectdreceiver] Migrate from OpenCensus data model to pdata

### PR DESCRIPTION
**Pull request by ZenoCC-Peng**
**Description:** <Describe what has changed.> 

Migrate from OpenCensus data model to pdata on collectdreceiver. Using golden to compare metrics for testing.  

Since wavefrontreceiver depends on collectdreceiver, adjustments are necessary in the go.mod and go.sum files.

 

**Link to tracking Issue:** <Issue number if applicable> [20760](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20760)

**Testing:** <Describe what testing was performed and which tests were added.> All test pass.

### CODE REVIEW

File: .chloggen/collectdreceiver-migrate-from-opensus-to-pdata.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: recevier/collectdreceiver
```
Comment (atoulme) [inline .chloggen/collectdreceiver-migrate-from-opensus-to-pdata.yaml:9]:
```suggestion
component: receiver/collectdreceiver
```

File: receiver/collectdreceiver/receiver_test.go [hunk 2]
```diff
@@ -97,7 +109,7 @@ func TestCollectDServer(t *testing.T) {
         "interval": 10.0,
         "plugin": "memory",
         "plugin_instance": "",
-        "time": 1415062577.4949999,
+        "time": 1415062577.494999900,
```
Comment (atoulme) [inline receiver/collectdreceiver/receiver_test.go:112]: this change is likely not needed.

**Review by atoulme — COMMENTED** [review]
Small nits. Good job!

File: receiver/collectdreceiver/receiver_test.go (hunk 2 above)
Comment (ZenoCC-Peng) [reply receiver/collectdreceiver/receiver_test.go]: Thanks, I changed.

File: receiver/collectdreceiver/collectd.go [hunk 3]
```diff
@@ -9,17 +9,22 @@ import (
...
 
+type TargetMetricType string
+
 const (
-	collectDMetricDerive   = "derive"
-	collectDMetricGauge    = "gauge"
-	collectDMetricCounter  = "counter"
-	collectDMetricAbsolute = "absolute"
+	GaugeMetricType      = TargetMetricType("gauge")
+	CumulativeMetricType = TargetMetricType("cumulative")
```
Comment (rmfitzpatrick) [inline receiver/collectdreceiver/collectd.go:22]: I think the convention is to not export unless we intend users to consume these. Also doesn't look like `GaugeMetricType` is used. Same applies to `TargetMetricType`, which may be unnecessary (derive and counter remain unexported strings for instance).

Comment (ZenoCC-Peng) [reply receiver/collectdreceiver/collectd.go]: Thanks!
That makes sense. 
I modified the code and not export them.

File: receiver/collectdreceiver/testdata/expected.yaml [hunk 4]
```diff
@@ -0,0 +1,180 @@
+resourceMetrics:
```
Comment (atoulme) [inline receiver/collectdreceiver/testdata/expected.yaml:1]: Is this file loaded in a test? I can't see it.

**Review by atoulme — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**